### ISSUE TITLE
State entity  invalid name for Cloud Connection 

### ISSUE THREAD

**Issue by pittcrew59** [PeteRager/lennoxs30#306]
I had this integration working and it stopped a month or more ago.  Just getting time to look at it and see a few errors in the log.  I have a M30 and noticed that I need to use Cloud.  I think I was local before.  I uninstalled and reinstalled so I could select Cloud. I put in my credentials and still getting errors.  See below:
===============================
Logger: homeassistant.config_entries
Source: config_entries.py:444
First occurred: 4:15:31 PM (2 occurrences)
Last logged: 4:31:57 PM

Error setting up entry the__p_tt_an_@y_ho_.c_m for lennoxs30
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 444, in async_setup
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/lennoxs30/__init__.py", line 344, in async_setup_entry
    await manager.s30_initialize()
  File "/config/custom_components/lennoxs30/__init__.py", line 538, in s30_initialize
    self.updateState(DS_CONNECTING)
  File "/config/custom_components/lennoxs30/__init__.py", line 504, in updateState
    self._hass.states.async_set(self.connection_state, state, self.getMetricsList(), force_update=True)
  File "/usr/src/homeassistant/homeassistant/core.py", line 1911, in async_set
    state = State(
            ^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 1430, in __init__
    raise InvalidEntityFormatError(
homeassistant.exceptions.InvalidEntityFormatError: Invalid entity id encountered: lennoxs30.conn_the__pittmans. Format should be <domain>.<object_id>

This error originated from a custom integration.

Logger: custom_components.lennoxs30
Source: custom_components/lennoxs30/__init__.py:706
Integration: Lennox S30/E30 (documentation, issues)
First occurred: February 9, 2024 at 10:44:53 AM (1787 occurrences)
Last logged: 4:32:06 PM

async_setup host [192.168.1.174] unexpected error Code [11] Reference [400] [login Client Connector Error - failed due host not reachable url [https://192.168.1.174/Endpoints/homeassistant/Connect]Message ConnectionKey(host='192.168.1.174', port=443, is_ssl=True, ssl=<ssl.SSLContext object at 0xffff5265cc50>, proxy=None, proxy_auth=None, proxy_headers_hash=None)Message [Errno 111] Connect call failed ('192.168.1.174', 443)]

**Comment by PeteRager** [comment]
The first error is a bug.

Either home assistant has tightened the rules for entity names or the integrations mangling of your email (for privacy) is producing and invalid name.  Either way that is a bug.

```
homeassistant.exceptions.InvalidEntityFormatError: Invalid entity id encountered: lennoxs30.conn_the__pittmans. Format should be <domain>.<object_id>
```

The second error is from the local connection being unable to reach the  M30 which is expected as the M30 is not reachable locally.

However, you should be able to work around the bug.

a) delete both integrations
b) restart HA (this is very important)
c) configure the cloud connection

Let me know if that gets rid of that error and if the integration comes up with data.

**Comment by pittcrew59** [comment]
Thanks Pete!  That worked.  I should have thought to restart after deleting the integration.  I really appreciate your help!
    On Saturday, February 10, 2024 at 05:46:21 PM CST, PeteRager ***@***.***> wrote:  
 
 


The first error is a bug.

Either home assistant has tightened the rules for entity names or the integrations mangling of your email (for privacy) is producing and invalid name. Either way that is a bug.
homeassistant.exceptions.InvalidEntityFormatError: Invalid entity id encountered: lennoxs30.conn_the__pittmans. Format should be <domain>.<object_id>

The second error is from the local connection being unable to reach the M30 which is expected as the M30 is not reachable locally.

However, you should be able to work around the bug.

a) delete both integrations
b) restart HA (this is very important)
c) configure the cloud connection

Let me know if that gets rid of that error and if the integration comes up with data.

**Comment by vsambaiah** [comment]
Hi Pete - I tried to follow the instructions above to delete the integration. Upon deletion, I still see it under downloaded integrations (HACS) - shows "Invalid Error" upon clicking it, but I don't see it under integrations. I have rebooted HA a few times. Seems like it is stuck in a limbo (purely due to my missteps perhaps). 

I have tried "re-downloading" and trying to configure from scratch with no success.

Help!

Thanks
Vish

![Lennox s30 HA Error](https://github.com/PeteRager/lennoxs30/assets/53830564/dea19d07-1a9b-40e6-9191-4f722081f380)

**Comment by PeteRager** [comment]
Since that panel comes up the integration looks like it's installed.  

What errors are in the log?

**Comment by vsambaiah** [comment]
Hi Pete - Nothing in the log. The last is from 6ish pm EST but not related to reconfiguration tries I did thereafter...

Logger: homeassistant.components.climate.const
Source: helpers/deprecation.py:205
Integration: Climate (documentation, issues)
First occurred: 6:57:20 PM (12 occurrences)
Last logged: 6:57:20 PM

• SUPPORT_TARGET_TEMPERATURE was used from myicomfort, this is a deprecated constant which will be removed in HA Core 2025.1. Use ClimateEntityFeature.TARGET_TEMPERATURE instead, please create a bug report at https://github.com/thevoltagesource/LennoxiComfort/issues
• SUPPORT_TARGET_TEMPERATURE_RANGE was used from myicomfort, this is a deprecated constant which will be removed in HA Core 2025.1. Use ClimateEntityFeature.TARGET_TEMPERATURE_RANGE instead, please create a bug report at https://github.com/thevoltagesource/LennoxiComfort/issues
• SUPPORT_AUX_HEAT was used from myicomfort, this is a deprecated constant which will be removed in HA Core 2025.1. Use ClimateEntityFeature.AUX_HEAT instead, please create a bug report at https://github.com/thevoltagesource/LennoxiComfort/issues
• SUPPORT_PRESET_MODE was used from myicomfort, this is a deprecated constant which will be removed in HA Core 2025.1. Use ClimateEntityFeature.PRESET_MODE instead, please create a bug report at



Regards
Vish

**Comment by PeteRager** [comment]
Please create a new issue as you are not seeing the error that was reported in this issue.

Next set of questions for that new jssue:
That other integration is for a different model Lennox thermostat.  What type of thermostat are you trying to connect to?  Has it ever worked?

Let's do a couple of things:
- take a screen shot of the integrations page, specifically I'd want to see if any Lennox integrations are configured. 
- post the contents of the custom_components/lennoxs30/manifest.json

**Comment by PeteRager** [comment]
Fixed in 2024.2.1
